deltachat-ffi/src/lib.rs

```diff
@@ -386,10 +388,19 @@ pub unsafe extern "C" fn dc_set_draft(
 ) {
     assert!(!context.is_null());
     let context = &*context;
+    let msg = if msg.is_null() { None } else { Some(&mut *msg) };
 
     dc_chat::dc_set_draft(context, chat_id, msg)
 }
 
+unsafe fn option_to_raw<'a, T>(x: Option<T>) -> *mut T {
```

Comment:
This seems like it'd be nicer as a trait?  Also a docstring with an example wouldn't be amiss.